### DISCUSSION TITLE
what makes commit hashes clickable?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [wezterm/wezterm#3618] [Q&A] [answered]
So I noticed that commit hashes from are indeed clickable. Clicking on one of the hashes from this command opens the respective commit at github which is awesome.

```bash
git log -n5 --pretty=format:'%h %s'
```
At first, I thought this was created by some hyperlink_rule, but the default rules contain nothing of the sort. 

Then I noticed that as soon as I pipe the result through something, the results are not clickable anymore, even though the output is unchanged
```bash
git log -n5 --pretty=format:'%h %s' | tail
```

So I assume that apparently, the clickability of the hashes is not implemented via some wezterm hyperlink rule, but rather by `git log`? Nevertheless, I also went through the [git log docs](https://git-scm.com/docs/git-log) and also could not find anything related there.

I thought about adding my own hyper_rule for this, but I am not sure how I can dynamically insert the git repo in there, so that the clickability of hashes works in different repos.

I tried the same thing in other terminals, and in none of them were git hashes clickable at all, so I assume the clickability is created by some wezterm feature, but I cannot really figure out which?

**Comment by wez** [comment]
There isn't anything built into wezterm to do this. From what you described, it sounds like `git` on your system is emitting OSC 8 hyperlinks when it knows that the output is a tty.  What version of git do you have installed? Is there some git configuration that does this on your system?

**Reply by chrisgrieser** [reply, marked as answer]
ah, okay, I have bisected my git config and actually figured out that an unrelated [delta](https://github.com/dandavison/delta) [configuration for grepping](https://dandavison.github.io/delta/grep.html) makes those hashes clickable. the git log results back through delta makes things clickable:

```bash
git log -n5 --pretty=format:'%h %s' | tail | delta
```

which is a pretty nifty trick. Thanks for the pointer that helped me figuring it out
